props/saber_fett263_buttons.h

```diff
@@ -293,166 +664,3131 @@ class SaberFett263Buttons : public virtual PropBase {
-      case EVENTID(BUTTON_BLADE_DETECT, EVENT_LATCH_ON, MODE_ANY_BUTTON | MODE_OFF):
-        // Might need to do something cleaner, but let's try this for now.
-        blade_detected_ = true;
-        FindBladeAgain();
-        SaberBase::DoBladeDetect(true);
-        return true;
+  void LengthPreview(int blade) {
+    show_length_.Start(blade);
+    length_edit_length = blade_length_ - 1;
+    ShowColorStyle::SetColor(GetColorArg(blade_num_, 1)); 
+  }
+ 
+  // Copy Color Arguments from one blade to another
+  void SetColorCopy() {
+    effect_num_ = 16;
+    char set_copy[16];
+    while (true) {
+      effect_num_ -= 1;
+      if (style_parser.UsesArgument(current_preset_.GetStyle(blade_num_), effect_num_ + 2)) break;
+    }
+    saved_color_ = GetColorArg(blade_num_, effect_num_);
+    hsl_ = saved_color_.toHSL();
+    NewColor(blade_num_, effect_num_);
+    current_preset_.Save();
+    effect_num_ = 0;
+  }
 
-      case EVENTID(BUTTON_BLADE_DETECT, EVENT_LATCH_OFF, MODE_ANY_BUTTON | MODE_ON):
+  // Set In/Out (Ignition/Retraction) Arguments 
+  void SetInOut() {
+    char style_arg[10];
+    // Reminder - use search and replace to rename "calc_" per PR comments
+    itoa(calc_, style_arg, 10);
+    current_preset_.SetStyle(blade_num_,style_parser.SetArgument(current_preset_.GetStyle(blade_num_), set_num_ + 2, style_arg));
+    current_preset_.Save();
+    switch (menu_type_) {
+      case MENU_STYLE_OPTION:
+      case MENU_IGNITION_TIME:
+      case MENU_IGNITION_OPTION:
+        UpdateFont(current_preset_.preset_num, false);
+        break;          
+      case MENU_PREON_OPTION:
+      case MENU_PREON_SIZE:
+        UpdateFont(current_preset_.preset_num, true);
+        break;
+      case MENU_RETRACTION_TIME:
+      case MENU_RETRACTION_OPTION:
+        UpdateStyle(current_preset_.preset_num);
+        off_event_ = true;
+        last_rotate_millis_ = millis();
+        break;
+      default:
+        break;
+    }
+  }
+
+  void RevertInOut() {
+    char style_arg[10];
+    itoa(arg_revert_, style_arg, 10);
+    current_preset_.SetStyle(blade_num_,style_parser.SetArgument(current_preset_.GetStyle(blade_num_), set_num_ + 2, style_arg));
+    switch (menu_type_) {
+      case MENU_RETRACTION_OPTION:
+      case MENU_RETRACTION_TIME:
+        char ig[10];
+        itoa(ignite_time_, ig, 10);
+        current_preset_.SetStyle(blade_num_,style_parser.SetArgument(current_preset_.GetStyle(blade_num_), 19, ig));
+        break;
+      default:
+        break;
+    }
+    current_preset_.Save();
+    UpdateFont(current_preset_.preset_num, false);
+  }
+
+  // Save IntArg values 16 ~ 25 from Edit Mode Menu selection
+  void SaveStyleSetting() {
+    char style_arg[10];
+    itoa(calc_, style_arg, 10);
+    current_preset_.SetStyle(blade_num_,style_parser.SetArgument(current_preset_.GetStyle(blade_num_), set_num_ + 2, style_arg));
+    current_preset_.Save();
+    // Stop location/size previews in Style Settings Mode
```

Comment:
This switch (and maybe some other code surrounding it) can be broken out into a helper function instead of repeated in the next function.

Reply:
broke out StopSettingPreview()